Add missing TORCH_CUDA_API annotation to throw_nccl_error

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31899 Revert "Move AutogradMeta and DeviceGuardImplInterface virtual methods out-of-line."
* #31162 Don't use RTLD_GLOBAL to load _C.
* #31161 Uniformly apply Windows logic in cpp_extensions everywhere
* #31236 Don't unconditionally compile runJITCPPTests
* #31176 Move AutogradMeta and DeviceGuardImplInterface virtual methods out-of-line.
* **#31157 Add missing TORCH_CUDA_API annotation to throw_nccl_error**
* #31155 Remove dead CAFFE2_LIBS variable
* #31152 Remove LibIRC logic from cmake.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262583](https://our.internmc.facebook.com/intern/diff/D19262583)

ezyang committed:
Add missing TORCH_CUDA_API annotation to throw_nccl_error

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit a9dcece:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a9dcece58e353880a3d1e8b9a6a02130a7a7d57c).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3915329) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** "pytorch android gradle build" ([details](https://dr.pytorch.org/build-details.html?build_id=72770762))
```
sys	0m0.169s
++ docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-a9dcece58e353880a3d1e8b9a6a02130a7a7d57c-android-x86_32
+ export id_x86_32=42fd1899624cdaf3d979f5d6753957f394b6a03813058c5b254243efed29686e
+ id_x86_32=42fd1899624cdaf3d979f5d6753957f394b6a03813058c5b254243efed29686e
+ export 'COMMAND=((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ COMMAND='((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ echo '((echo' '"source' './workspace/env"' '&&' echo '"sudo' chown -R jenkins 'workspace")' '|' docker exec -u jenkins -i '"$id_x86_32"' 'bash)' '2>&1'
+ unbuffer bash ./command.sh
+ ts
+ docker pull 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-a9dcece58e353880a3d1e8b9a6a02130a7a7d57c-android-arm-v7a
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-a9dcece58e353880a3d1e8b9a6a02130a7a7d57c-android-arm-v7a not found
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang committed:
Update on "Add missing TORCH_CUDA_API annotation to throw_nccl_error"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Add missing TORCH_CUDA_API annotation to throw_nccl_error"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Add missing TORCH_CUDA_API annotation to throw_nccl_error"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Add missing TORCH_CUDA_API annotation to throw_nccl_error"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262583](https://our.internmc.facebook.com/intern/diff/D19262583)

[ghstack-poisoned]
